SetDefaultSinkId/explainer.md

```diff
@@ -0,0 +1,184 @@
+# Set Default Audio Output Device
+
+Author: [Sunggook Chue] [Ravikiran Ramachandra] [Steve Becker] [Andy Luhrs]
+
+## Status of this Document
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+
+This document status: Active
+Expected venue: [WebRTC WG](https://www.w3.org/groups/wg/webrtc)
+Current version: this document
+
+## Introduction
+
+Setting Audio Output Device In Frames
+
+In embedded web experiences, the top-level frame grants iframes access to system media devices like microphones, cameras and speakers. This is typically done through the permission policy (https://developer.mozilla.org/en-US/docs/Web/HTTP/Permissions_Policy) in the iframe tag. However, this access comes with two key challenges:
+
+- Independent Choice: Each iframe independently chooses its own media device. The top-level frame cannot directly influence or view this selection due to browser security restrictions (cross-origin boundary).
+- Unsynchronized Changes: When the top-level frame changes its media device, the iframes remain unaware unless they communicate using methods like postMessage. This lack of automatic synchronization can lead to inconsistencies and a disjointed user experience.
+
+We’d like to propose a new API to set the default audio output device for the current top frame and of its sub frames.
+
+## Goals
+
+- Developers can modify the default audio output for the top-level frame page.
+- Developers can modify the default audio output for the sub frame pages.
+- Developers can modify the default audio output for the top-level frame and all other sub frames from the same-origin sub frame.
+- Any media element or audio context can continue to override this default setting using the existing setSinkId API.
+
+## Non Goals
+- A change notification for default audio output is not in scope of this project.
```

Comment:
Media Elements have a sinkId property, which might provide a way to detect change notifications for the default audio output device.  How does this proposal integrate with the Media Element's sinkId property?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/sinkId

Web Audio has the same property as well:

https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/sinkId

Reply:
It will continue to return default device id ("") unless component call setSinkId with custom deviceId because the proposed API change the default audio device output. I will add it on the non-goals.